tests/DCPS/Compiler/xcdr/xcdr.cpp

```diff
@@ -476,6 +476,54 @@ TEST(mutable_tests, to_additional_field_xcdr2_test)
   /// value?) if we decide on that.
 }
 
+TEST(mutable_tests, to_additional_field_mustunderstatnd_test)
+{
+  const unsigned char expected[] = {
+    // Delimiter
+    0x00, 0x00, 0x00, 0x2a, // +4 = 4
+    // long_field
+    0x00, 0x00, 0x00, 0x06, // PID +4 = 8
+    0x7f, 0xff, 0xff, 0xff, // +4 = 12
+    // additional_field @must_understand
+    0x80, 0x00, 0x00, 0x01, // PID +4 = 16
+    0x12, 0x34, 0x56, 0x78, // +4 = 20
+    // long_long_field
+    0x00, 0x00, 0x00, 0x0a, // PID +4 = 24
+    0x7f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, // +8 = 32
+    // octet_field
+    0x00, 0x00, 0x00, 0x08, // PID +4 = 36
+    0x01, // +1 = 37
+    // short_field
+    0x00, 0x00, 0x00, // +3 pad = 40
+    0x00, 0x00, 0x00, 0x04, // PID +4 = 44
+    0x7f, 0xff // +2 = 46
+  };
+
+  // Deserialization should fail for unknown must_understand field
+  {
+    ACE_Message_Block buffer(1024);
+    buffer.copy((const char*)expected, sizeof(expected));
+    Serializer serializer(&buffer, xcdr2);
+
+    MutableStruct result;
+    EXPECT_FALSE(serializer >> result);
+  }
+
+  // Deserialize and Compare C++ Values
+  {
+    ACE_Message_Block buffer(1024);
+    buffer.copy((const char*)expected, sizeof(expected));
+    Serializer serializer(&buffer, xcdr2);
+
+    AdditionalFieldMutableStruct result;
+    EXPECT_TRUE(serializer >> result);
```

Comment:
Looking at this I'm thinking this should probably be:

```suggestion
    ASSERT_TRUE(serializer >> result);
```

In Google test, this stops the test, where `EXPECT_*` keeps going. I'm thinking this because the value comparison below shouldn't happen if deserialization failed, but I know this is copied from my general test code, so it's the same problem through out the file.

Reply:
Based on the behaviors of EXPECT_TRUE and ASSERT_TRUE, your suggested change makes perfect sense. Similar changes were also made in two other places (Lines 142 & 149).